pool/test/message_template_test.ml

```diff
@@ -109,6 +109,138 @@ let create_invitation language ?entity_uuid () =
+  let find_template = find_template_to_send experiment in
+  let%lwt default_template_de = find_default_by_label_and_language de label in
+  let%lwt default_template_en = find_default_by_label_and_language en label in
+  let%lwt res_de = find_template contact_de in
+  let%lwt res_en = find_template contact_en in
+  (* As no experiment language is defined, always expect a template in the
+     contat language to be returned *)
+  let () =
+    Alcotest.(
+      check Test_utils.message_template "succeeds" default_template_de res_de)
+  in
+  let () =
+    Alcotest.(
+      check Test_utils.message_template "succeeds" default_template_en res_en)
+  in
+  Lwt.return_unit
+;;
+
+let get_template_without_experiment_language _ () =
+  let open LanguageTestsData in
+  let label = invitation_label in
+  let experiment = create_experiment () in
+  let find_template = find_template_to_send experiment in
+  let%lwt template_de =
+    create_experiment_message_template experiment label de
+  in
+  let%lwt default_template_en = find_default_by_label_and_language en label in
+  let%lwt res_de = find_template contact_de in
+  let%lwt res_en = find_template contact_en in
+  (* Expect the custom de template and the default en tempalte to be returned *)
+  let () =
+    Alcotest.(check Test_utils.message_template "succeeds" template_de res_de)
+  in
+  let () =
+    Alcotest.(
+      check Test_utils.message_template "succeeds" default_template_en res_en)
+  in
+  Lwt.return_unit
+;;
+
+let get_template_with_experiment_language _ () =
+  let open LanguageTestsData in
+  let label = invitation_label in
+  let experiment = create_experiment ~language:Pool_common.Language.De () in
+  let find_template = find_template_to_send experiment in
+  let%lwt default_template_de = find_default_by_label_and_language de label in
+  let%lwt res_de = find_template contact_de in
+  let%lwt res_en = find_template contact_en in
+  (* Always expect the template to be in experiment language *)
+  let () =
+    Alcotest.(
+      check Test_utils.message_template "succeeds" default_template_de res_de)
+  in
+  let () =
+    Alcotest.(
+      check Test_utils.message_template "succeeds" default_template_de res_en)
+  in
+  Lwt.return_unit
+;;
+
+let get_template_with_experiment_language_and_template _ () =
+  let open LanguageTestsData in
+  let label = invitation_label in
+  let experiment = create_experiment ~language:Pool_common.Language.De () in
+  let%lwt template_de =
+    create_experiment_message_template experiment label de
+  in
+  let find_template = find_template_to_send experiment in
+  let%lwt res_de = find_template contact_de in
+  let%lwt res_en = find_template contact_en in
+  (* Always expect the template to be in experiment language *)
+  let () =
+    Alcotest.(check Test_utils.message_template "succeeds" template_de res_de)
+  in
+  let () =
+    Alcotest.(check Test_utils.message_template "succeeds" template_de res_en)
+  in
```

Comment:
Define a partially applied function called `check_experiment_message_template` with one input argument for the English and German contact. With this, the previous comment becomes obsolete.